pkg/lib/authn/authenticator/password/generator.go

```diff
@@ -177,47 +173,70 @@ func (g *Generator) generate() (string, error) {
 	return shuffled, nil
 }
 
-func createSetFromList(list string) string {
-	var set strings.Builder
-	setMap := make(map[byte]bool)
+func (g *Generator) prepareCharList() string {
```

Comment:
Change this to be a private function `func prepareCharacterSet(policy) string` and write tests for it.